utils/cp_parallel.py

```diff
@@ -0,0 +1,176 @@
+
+
+def results_to_log(
+    results: List[subprocess.CompletedProcess], log_dir: pathlib.Path, run_name: str
+) -> None:
+    """
+    This function converts the list of subprocess results from a CellProfiler parallelization run
+    into a separate log file for each process.
+
+    Args:
+        results (List[subprocess.CompletedProcess]): Outputs from subprocess.run
+        log_dir (pathlib.Path): Directory for log files
+        run_name (str): Name for the type of CellProfiler run (e.g., whole image features)
+    """
+    for result in results:
+        # Extract plate name and decoded output
+        plate_name = result.args[6].name
+        output_string = result.stderr.decode("utf-8")
+
+        # Set the log file path
+        log_file_path = log_dir / f"{plate_name}_{run_name}_run.log"
+
+        # Create a new logger instance for each plate
+        logger = logging.getLogger(plate_name)
+        logger.setLevel(logging.INFO)
+
+        # Create a file handler for the logger
+        handler = logging.FileHandler(log_file_path)
+        formatter = logging.Formatter(
+            "[%(asctime)s] [Process ID: %(process)d] %(message)s"
+        )
+        handler.setFormatter(formatter)
+
+        # Add the handler to the logger
+        logger.addHandler(handler)
+
+        # Log the output
+        logger.info(f"Plate Name: {plate_name}")
+        logger.info(f"Output String: {output_string}")
+
+        # Close the handler after logging to avoid resource leaks
+        handler.close()
+        logger.removeHandler(handler)
+
+
+def run_cellprofiler_parallel(
+    plate_info_dictionary: dict,
+    run_name: str,
+) -> None:
+    """
+    This function utilizes multi-processing to run CellProfiler pipelines in parallel.
+
+    Args:
+        plate_info_dictionary (dict): dictionary with all paths for CellProfiler to run a pipeline
+        run_name (str): a given name for the type of CellProfiler run being done on the plates (example: whole image features)
+
+    Raises:
+        FileNotFoundError: if paths to pipeline and images do not exist
+    """
+    # create a list of commands for each plate with their respective log file
+    commands = []
+
+    # make logs directory
+    log_dir = pathlib.Path("./logs")
+    os.makedirs(log_dir, exist_ok=True)
+
+    # iterate through each plate in the dictionary
+    for _, info in plate_info_dictionary.items():
+        # set paths for CellProfiler
+        path_to_pipeline = info["path_to_pipeline"]
+        path_to_output = info["path_to_output"]
+
+        # make output directory if it is not already created
+        pathlib.Path(path_to_output).mkdir(exist_ok=True)
+
+        # check to make sure paths to pipeline are correct before running the pipeline
+        if not pathlib.Path(path_to_pipeline).resolve(strict=True):
+            raise FileNotFoundError(
+                f"The file '{pathlib.Path(path_to_pipeline).name}' does not exist"
+            )
```

Comment:
Not sure what you mean here. This is not related to the custom error and this specific error should still be raised since it is after the path is loaded in from the directory.